File: 프로그래머스/01/약수의 갯수와 덧셈/02.py
# 약수의 갯수가 짝수이면 더한다.

from math import sqrt as sq

def solution(n1,n2):
   ans = 0
   for i in range(n1,n2+1):
      if sq(i) == int(sq(i)):
         ans -= i
      else:
         ans += i
   return ans

print(solution(13,17))

File: 프로그래머스/02/네트워크/02.js
let cpComputers;
let visited;
function solution(n, computers) {
  let ans = 0;
  cpComputers = computers;
  visited = new Array(n).fill(false);

  for (const idx in cpComputers) {
    ans += dfs(idx);
  }
  return ans;
}

function dfs(idx) {
  if (visited[idx] === true) return 0;
  visited[idx] = true;
  for (const i in cpComputers[idx]) {
    if (cpComputers[idx][i] == 1) {
      dfs(i);
    }
  }
  return 1;
}
